pkg/cfn/manager/waiters.go

```diff
@@ -1,121 +1,45 @@
 package manager
```

Comment:
Nice, really like the waiter refactor.

Reply:
Wait, your description says:

```
migrate the waiters (separate commit)
```

But this has waiter refactors. :D I thought separate commit meant separate PR. :) But maybe I was just mistaken :)

Reply:
yeah separate commit within the same PR. Idea being you can review the commit with all the types/contexts passed around, and then the actually "work" of waiters separately